x-pack/legacy/plugins/siem/public/pages/hosts/index.tsx

```diff
@@ -44,205 +36,51 @@ export const HostsContainer = React.memo<Props>(({ url }) => (
-                  <HostDetailsBody
-                    deleteQuery={deleteQuery}
-                    from={from}
-                    to={to}
-                    setQuery={setQuery}
-                    isInitializing={isInitializing}
-                    detailName={props.match.params.detailName}
-                    children={AuthenticationsQueryTabBody}
-                  />
-                )}
-              />
-              <Route
-                path={`${hostsPagePath}/:detailName/:tabName(${HostsTableType.uncommonProcesses})`}
-                render={() => (
-                  <HostDetailsBody
-                    deleteQuery={deleteQuery}
-                    from={from}
-                    to={to}
-                    setQuery={setQuery}
-                    isInitializing={isInitializing}
-                    detailName={props.match.params.detailName}
-                    children={UncommonProcessQueryTabBody}
-                  />
-                )}
-              />
-              <Route
-                path={`${hostsPagePath}/:detailName/:tabName(${HostsTableType.anomalies})`}
-                render={() => (
-                  <HostDetailsBody
-                    deleteQuery={deleteQuery}
-                    from={from}
-                    to={to}
-                    setQuery={setQuery}
-                    isInitializing={isInitializing}
-                    detailName={props.match.params.detailName}
-                    children={AnomaliesQueryTabBody}
-                  />
-                )}
-              />
-              <Route
-                path={`${hostsPagePath}/:detailName/:tabName(${HostsTableType.events})`}
-                render={() => (
-                  <HostDetailsBody
-                    from={from}
-                    to={to}
-                    setQuery={setQuery}
-                    isInitializing={isInitializing}
-                    detailName={props.match.params.detailName}
-                    children={EventsQueryTabBody}
-                  />
-                )}
-              />
-            </>
+            <HostDetails
+              hostDetailsPagePath={hostDetailsPagePath}
+              detailName={props.match.params.detailName}
+              from={from}
+              to={to}
+              setQuery={setQuery}
+              isInitializing={isInitializing}
+              deleteQuery={deleteQuery}
+            />
           )}
         />
         <Route
           path={`${url}/:detailName`}
-          render={({ location: { search = '' } }) => (
-            <Redirect
-              from={`${url}/:detailName`}
-              to={`${url}/:detailName/${HostsTableType.authentications}${search}`}
-            />
+          render={({
+            match: {
+              params: { detailName },
+            },
+            location: { search = '' },
+          }) => (
+            <Redirect to={`${url}/:${detailName}/${HostsTableType.authentications}${search}`} />
```

Comment:
Same suggestion as the one above.

Reply:
We're not able to use path definition in redirect as it expects to get the exact path

Reply:
ok, no problem. My bad :p